pkg/controllers/user/istio/metricExpression.go

```diff
@@ -0,0 +1,555 @@
+package istio
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	managementv3 "github.com/rancher/types/apis/management.cattle.io/v3"
+)
+
+func getPredefinedIstioMetrics() []*managementv3.MonitorMetric {
+	yamls := strings.Split(IstioMetricsTemplate, "\n---\n")
+	var rtn []*managementv3.MonitorMetric
+	for _, yml := range yamls {
+		var tmp managementv3.MonitorMetric
+		if err := yamlToObject(yml, &tmp); err != nil {
+			panic(err)
+		}
+		if tmp.Name == "" {
+			continue
+		}
+		rtn = append(rtn, &tmp)
+	}
+
+	return rtn
+}
+
+func getPredefinedIstioClusterGraph() []*managementv3.ClusterMonitorGraph {
+	yamls := strings.Split(IstioClusterGraphTemplate, "\n---\n")
+	var rtn []*managementv3.ClusterMonitorGraph
+	for _, yml := range yamls {
+		var tmp managementv3.ClusterMonitorGraph
+		if err := yamlToObject(yml, &tmp); err != nil {
+			panic(err)
+		}
+		if tmp.Name == "" {
+			continue
+		}
+		rtn = append(rtn, &tmp)
+	}
+
+	return rtn
+}
+
+func getPredefinedIstioProjectGraph() []*managementv3.ClusterMonitorGraph {
+	yamls := strings.Split(IstioProjectGraphTemplate, "\n---\n")
+	var rtn []*managementv3.ClusterMonitorGraph
+	for _, yml := range yamls {
+		var tmp managementv3.ClusterMonitorGraph
+		if err := yamlToObject(yml, &tmp); err != nil {
+			panic(err)
+		}
+		if tmp.Name == "" {
+			continue
+		}
+		rtn = append(rtn, &tmp)
+	}
+
+	return rtn
+}
+
+func yamlToObject(yml string, obj interface{}) error {
+	jsondata, err := yaml.YAMLToJSON([]byte(yml))
+	if err != nil {
+		return err
+	}
+	if err := json.Unmarshal(jsondata, obj); err != nil {
+		return err
+	}
+	return nil
+}
+
+var (
+	IstioMetricsTemplate = `
```

Comment:
this seems weird, next week, id like to discuss what it is and why its needed, but i can merge as-is, not a blocker

Reply:
got it